third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -2324,6 +2542,21 @@ impl Frame {
                             .operand_stack
                             .push(Value::struct_(Struct::pack(args)))?;
                     },
+                    Bytecode::PackVariant(idx) => {
+                        let info = resolver.get_struct_variant_at(*idx);
+                        let struct_type = resolver.create_struct_type(&info.definition_struct_type);
+                        check_depth_of_type(resolver, &struct_type)?;
+                        gas_meter.charge_pack(
```

Comment:
I see you use `charge_pack` here, should we have dedicated gas metering methods for variants? Because otherwise here we use gas parameters for general pack for structs.